umm-spec-lib/src/cmr/umm_spec/validation/data_date.clj

```diff
@@ -0,0 +1,43 @@
+(ns cmr.umm-spec.validation.data-date
+  "Defines validations for UMM collection DataDates and MetadataDates"
+  (:require
+   [clj-time.core :as time]
+   [cmr.umm-spec.date-util :as date-util]))
+
+(defn data-dates-warning-validation
+  "Validate that DataDates/MetadataDates' 
+   CREATE, UPDATE, REVIEW and DELETE date values are in the right order and range: 
+   CREATE needs to be <= UPDATE and they both need to be in the past. 
+   REVIEW needs to be <= DELETE and they both need to be in the future.
+   field-path is :DataDates/:MetadataDates, 
+   value is the value of :DataDates/:MetadataDates."
+  [field-path value]
+  (let [latest-create (date-util/latest-date-of-type value "CREATE")
+        earliest-update (date-util/earliest-date-of-type value "UPDATE")
+        latest-update (date-util/latest-date-of-type value "UPDATE")
+        earliest-review (date-util/earliest-date-of-type value "REVIEW")
+        latest-review (date-util/latest-date-of-type value "REVIEW")
+        earliest-delete (date-util/earliest-date-of-type value "DELETE")
+        warning-msgs (concat 
```

Comment:
I understand your point.  If I could completely separate the validation of CREATE, UPDATE, REVIEW and DELETE, I would do that. But since I still have to validate the relative relationships, the detailed function names could be a little confusing. Given the length of this function, which is very short, I really don't feel it's necessary to break it down further.  In general, I feel that you can almost always break down a function further based on some kind of meaningful conceptions, depending on how you look at it and how far you want to break it...